### PR TITLE
Address endpoint

### PR DESCRIPTION
**Pull request by j-kon**
**What does this PR do?**

A summary of what this PR is about

**List of task completed**

-   [x] Specify what tasks are completed in this PR
-   [ ] Specify what's left to be done

**How can this be tested?**

- Highlight how this PR can be tested

**Additional info**

Add more info if any. eg fixes #.

### CODE REVIEW

File: Frontend/lib/controllers/auth_controller.dart [hunk 1]
```diff
@@ -13,38 +13,58 @@ class AuthenticationController extends GetxController {
...
     update();
-    const endPoint = '$baseUrl/register/';
-    final response = await _coSignApi.post(endPoint, body: signUpDetails);
-    if (response?.statusCode == 201) {
-      userRegister = UserRegister.fromJson(response!.data);
+    try {
+      const endPoint = '$baseUrl/register/';
+
+      final response = await _coSignApi.post(endPoint, body: signUpDetails);
+      print(response);
```
Comment (toshmanuel) [inline Frontend/lib/controllers/auth_controller.dart]: kindly remove

Comment (toshmanuel) [inline Frontend/lib/controllers/auth_controller.dart]:
```suggestion
```

File: Frontend/lib/controllers/address_controller.dart [hunk 2]
```diff
@@ -22,21 +23,23 @@ class AddressController extends GetxController {
...
+      final generateAddresses = GenerateAddresses.fromJson(response!.data);
+      final address = generateAddresses.address;
       isLoading = false;
-      update();
-      return Future.value(true);
+
+      return Future.value(address);
     }
-    isLoading = true;
-    return Future.value(false);
```
Comment (Muneerah1612) [inline Frontend/lib/controllers/address_controller.dart:39]: just curious, what is the difference between "Future.value(false)" and "Future.value('')" ?

File: Frontend/lib/controllers/address_controller.dart [hunk 3]
```diff
@@ -67,6 +70,21 @@ class AddressController extends GetxController {
...
+    print(response);
+    if (response?.statusCode == 200) {
+      isLoading = false;
+      update();
+      totalUTXO = TotalUTXO.fromJson(response!.data);
+    }
+    return totalUTXO;
+  }
+
   getNewAddressQuery() async {
```
Comment (Muneerah1612) [inline Frontend/lib/controllers/address_controller.dart]: Have you consumed this?

File: Frontend/lib/controllers/auth_controller.dart (hunk 1 above)
Comment (Muneerah1612) [inline Frontend/lib/controllers/auth_controller.dart]: You can remove this, since you used it to debug

File: Frontend/lib/data/models/user_register.dart [hunk 4]
```diff
@@ -1,21 +1,15 @@
 class UserRegister {
   String? email;
-  String? password;
```
Comment (Muneerah1612) [inline Frontend/lib/data/models/user_register.dart:3]: I can see that you removed everything that has to do with password, why is that?

Comment (toshmanuel) [inline Frontend/lib/data/models/user_register.dart:3]: Why did you remove the passwords?
Are they not meant to be sent to the server for a successful registration?

File: Frontend/lib/screens/auth/login_screen.dart [hunk 5]
```diff
@@ -24,14 +25,26 @@ class _LoginScreenState extends State<LoginScreen> {
...
   }
 
   bool _obscureText = true;
+
+  @override
+  void initState() {
+    super.initState();
+
+    //Todo: Remove this
+    if (kDebugMode) {
```
Comment (Muneerah1612) [inline Frontend/lib/screens/auth/login_screen.dart:42]: Please can you explain this? if it is not required, can you remove it then?

File: Frontend/lib/data/models/user_register.dart (hunk 4 above)
Comment (toshmanuel) [reply Frontend/lib/data/models/user_register.dart:3]: I went through the files. I didn't see where the data with the passwords were being sent to the server.

**Review by toshmanuel — CHANGES_REQUESTED** [review]

**Review by Muneerah1612 — COMMENTED** [review]
Please, always add a new line at the end of the files

File: Frontend/lib/controllers/auth_controller.dart (hunk 1 above)
Comment (j-kon) [reply Frontend/lib/controllers/auth_controller.dart]: I will remove it, just using print to see the progress on my terminal

Comment (j-kon) [reply Frontend/lib/controllers/auth_controller.dart]: > 

I don’t really understand your comment here

File: Frontend/lib/data/models/user_register.dart (hunk 4 above)
Comment (j-kon) [reply Frontend/lib/data/models/user_register.dart:3]: > Why did you remove the passwords? Are they not meant to be sent to the server for a successful registration?

I discussed how I auto generate my models which I once showed you, I generate models from example response

File: Frontend/lib/controllers/address_controller.dart (hunk 2 above)
Comment (j-kon) [reply Frontend/lib/controllers/address_controller.dart:39]: > just curious, what is the difference between "Future.value(false)" and "Future.value('')" ?

The future value was bool, which accept false and true while the other is string which I make the string empty.

File: Frontend/lib/controllers/address_controller.dart (hunk 3 above)
Comment (j-kon) [reply Frontend/lib/controllers/address_controller.dart]: > Have you consumed this?

Yes I have

File: Frontend/lib/controllers/auth_controller.dart (hunk 1 above)
Comment (j-kon) [reply Frontend/lib/controllers/auth_controller.dart]: I will remove it

File: Frontend/lib/data/models/user_register.dart (hunk 4 above)
Comment (j-kon) [reply Frontend/lib/data/models/user_register.dart:3]: > I can see that you removed everything that has to do with password, why is that?

I discussed how I auto generate my models which I once showed you, I generate models from example response

File: Frontend/lib/screens/auth/login_screen.dart (hunk 5 above)
Comment (j-kon) [reply Frontend/lib/screens/auth/login_screen.dart:42]: The app is still on debug mode, it make work faster and I will remove it during clean up.

File: Frontend/lib/controllers/auth_controller.dart (hunk 1 above)
Comment (toshmanuel) [reply Frontend/lib/controllers/auth_controller.dart]: okay

File: Frontend/lib/data/models/user_register.dart (hunk 4 above)
Comment (toshmanuel) [reply Frontend/lib/data/models/user_register.dart:3]: But in example response, there is password there

File: Frontend/lib/screens/auth/login_screen.dart (hunk 5 above)
Comment (Muneerah1612) [reply Frontend/lib/screens/auth/login_screen.dart:42]: okay, I am asking this because we have been using this same email and password to login, do you mind if we jump on a call in the morning to try something else?

File: Frontend/lib/controllers/auth_controller.dart (hunk 1 above)
Comment (toshmanuel) [reply Frontend/lib/controllers/auth_controller.dart]: This is a suggestion for the above comment

File: Frontend/lib/screens/auth/login_screen.dart (hunk 5 above)
Comment (j-kon) [reply Frontend/lib/screens/auth/login_screen.dart:42]: Sure, by 9am

Comment (Muneerah1612) [reply Frontend/lib/screens/auth/login_screen.dart:42]: Alright,till then

File: Frontend/lib/data/models/user_register.dart (hunk 4 above)
Comment (Muneerah1612) [reply Frontend/lib/data/models/user_register.dart:3]: Okay, I get that. Looking at this, it seems you removed it because it was there before

Comment (toshmanuel) [reply Frontend/lib/data/models/user_register.dart:3]: again, from this commit it looks like those fields were removed from the model not generated

Comment (j-kon) [reply Frontend/lib/data/models/user_register.dart:3]: > Okay, I get that. Looking at this, it seems you removed it because it was there before

Yeah that was a wrong auto generating

Comment (j-kon) [reply Frontend/lib/data/models/user_register.dart:3]: I removed is it because I was generating the wrong response.

**Review by Muneerah1612 — APPROVED** [review]
Looks good to me, just make sure you always add a new line at the end of a file

**Review by toshmanuel — APPROVED** [review]
